gen/schema/schema.yaml

```diff
@@ -7,7 +7,8 @@ rest_endpoint: str(required=False) # REST endpoint path
 put_create: bool(required=False) # Set to true if the PUT request is used for create
 no_update: bool(required=False) # Set to true if the PUT request is not supported
 no_delete: bool(required=False) # Set to true if the DELETE request is not supported
-data_source_name_query: bool(required=False) # Set to true if the data source supports name queries
+data_source_name_query: bool(required=False) # Set to true if the data source supports name queries ### Equal to "query_parameter: name"
+query_parameter: str(required=False) # Additional parameter (tf_name), in addition to `id`, to search for object in `data source` and `put create`
```

Comment:
In other implementations `query_parameter` is referring to a query param in the URL. I would suggestion to use something like `data_source_query_attribute` to be more specific.

Also, would it make sense to define this at the attribute instead of definition level? Which would possibly also simplify the implementation (templates). Something like:

```yaml
attributes:
  - model_name: ifname
    data_path: [interface]
    tf_name: interface_logical_name
    description: Logical Name of the interface of BFD assignment if SINGLE_HOP selected.
    type: String
    data_source_query: true
    example: "outside"
```

Reply:
Sure, let's rename it to `data_source_query_attribute`.
I believe making this global, makes it easy to find. Also putting it under attribute as `true` value, could suggest that we can have multiple occurrences of those, which is not our intention at the moment.

Reply:
Implemented under attribute as suggested